pkgs/flojoy/flojoy/flojoy_cloud.py

```diff
@@ -1,475 +1,344 @@
-
-        return response
-
-    def list_measurements(self, size: int = 10) -> list:
-        """
-        A method that lists the number of measurements specified.
-
-        If the number of measurements is less than the specified number,
-        an error will be thrown.
-        """
-        url = f"{self.base_url}/measurements/?size={size}"
-        response = requests.request("GET", url, headers=self.headers)
-        response = json.loads(response.text)
-        response = response["data"]
-
-        return response
-
-    def fetch_measurement(self, meas_id: str):
-        """
-        A method fetchs measurements from the client.
-        """
-        url = f"{self.base_url}/measurements/{meas_id}"
-        response = requests.request("GET", url, headers=self.headers)
-        response = json.loads(response.text)
-
-        return response
-
-    def rename_measurement(self, meas_id: str, name: str):
-        """
-        Rename the specified measurement.
-        """
-        url = f"{self.base_url}/measurements/{meas_id}"
-        payload = payload = json.dumps({"name": name})
-        response = requests.request("PATCH", url, headers=self.headers, data=payload)
-        response = json.loads(response.text)
-
-        return response
-
-    def store_dc(self, data, dc_type: str, meas_id: str, name: str):
-        """
-        A method that stores a formatted data payload in a measurement.
-        """
-        url = f"{self.base_url}/dcs/add/{meas_id}"
-        payload = self._create_payload(data, dc_type, name)
-        response = requests.request("POST", url, headers=self.headers, data=payload)
-
-        return json.loads(response.text)
+                return model.model_validate_json(json_str)
+
+    @overload
+    def __req(
+        self,
+        method: HttpMethod,
+        endpoint: str,
+        *,
+        query_params: dict[str, int | str] | None = None,
+        data: str | None = None,
+        body: dict | None = None,
+        parse_as: type[T],
+    ) -> T:
+        ...
+
+    @overload
+    def __req(
+        self,
+        method: HttpMethod,
+        endpoint: str,
+        *,
+        query_params: dict[str, int | str] | None = None,
+        data: str | None = None,
+        body: dict | None = None,
+        parse_as: TypeAdapter[U],
+    ) -> U:
+        ...
+
+    @overload
+    def __req(
+        self,
+        method: HttpMethod,
+        endpoint: str,
+        *,
+        query_params: dict[str, int | str] | None = None,
+        data: str | None = None,
+        body: dict | None = None,
+        parse_as: None = None,
+    ) -> None:
+        ...
+
+    def __req(
+        self,
+        method: HttpMethod,
+        endpoint: str,
+        *,
+        query_params: dict[str, int | str] | None = None,
+        data: str | None = None,
+        body: dict | None = None,
+        parse_as: type[T] | TypeAdapter[U] | None = None,
+    ) -> T | U | None:
+        res = requests.request(
```

Comment:
Should probably use httpx instead of requests imo. 
https://www.python-httpx.org/